### PR TITLE
fix: fix replication failed SP err recording issue

### PR DESCRIPTION
**Pull request by alexgao001**
### Description

This pr fixes when replication failed, the abnormal secondary sp's index and err info might got overridden by other normal SP's context canceled err .

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### CODE REVIEW

File: base/gfspapp/authenticator_server.go [hunk 1]
```diff
@@ -11,10 +12,20 @@ import (
...
+var (
+	ErrAuthenticatorTaskDangling = gfsperrors.Register(BaseCodeSpace, http.StatusBadRequest, 990101, "OoooH... request lost")
+)
+
 var _ gfspserver.GfSpAuthenticationServiceServer = &GfSpBaseApp{}
 
 func (g *GfSpBaseApp) GfSpVerifyAuthentication(ctx context.Context, req *gfspserver.GfSpAuthenticationRequest) (
 	*gfspserver.GfSpAuthenticationResponse, error) {
+	if req == nil {
+		log.Error("failed to ask authenticator due to pointer dangling")
```
Comment (will-2012) [inline base/gfspapp/authenticator_server.go:24]: Refine this log.

File: base/gfspapp/authenticator_server.go [hunk 2]
```diff
@@ -38,6 +49,12 @@ func (g *GfSpBaseApp) GfSpVerifyAuthentication(ctx context.Context, req *gfspser
 
 // GetAuthNonce get the auth nonce for which the Dapp or client can generate EDDSA key pairs.
 func (g *GfSpBaseApp) GetAuthNonce(ctx context.Context, req *gfspserver.GetAuthNonceRequest) (*gfspserver.GetAuthNonceResponse, error) {
+	if req == nil {
+		log.Error("failed to ask authenticator due to pointer dangling")
```
Comment (will-2012) [inline base/gfspapp/authenticator_server.go:53]: Ditto.

File: base/gfspapp/authenticator_server.go [hunk 3]
```diff
@@ -58,6 +75,12 @@ func (g *GfSpBaseApp) GetAuthNonce(ctx context.Context, req *gfspserver.GetAuthN
 
 // UpdateUserPublicKey updates the user public key once the Dapp or client generates the EDDSA key pairs.
 func (g *GfSpBaseApp) UpdateUserPublicKey(ctx context.Context, req *gfspserver.UpdateUserPublicKeyRequest) (*gfspserver.UpdateUserPublicKeyResponse, error) {
+	if req == nil {
+		log.Error("failed to ask authenticator due to pointer dangling")
```
Comment (will-2012) [inline base/gfspapp/authenticator_server.go:79]: Ditto.

File: base/gfspapp/authenticator_server.go [hunk 4]
```diff
@@ -71,6 +94,12 @@ func (g *GfSpBaseApp) UpdateUserPublicKey(ctx context.Context, req *gfspserver.U
 
 // VerifyGNFD1EddsaSignature verifies the signature signed by user's EDDSA private key.
 func (g *GfSpBaseApp) VerifyGNFD1EddsaSignature(ctx context.Context, req *gfspserver.VerifyGNFD1EddsaSignatureRequest) (*gfspserver.VerifyGNFD1EddsaSignatureResponse, error) {
+	if req == nil {
+		log.Error("failed to ask authenticator due to pointer dangling")
```
Comment (will-2012) [inline base/gfspapp/authenticator_server.go:98]: Ditto.

**Review by will-2012 — APPROVED** [review]

File: base/gfspapp/authenticator_server.go (hunk 4 above)
Comment (alexgao001) [reply base/gfspapp/authenticator_server.go]: refined.

File: base/gfspapp/authenticator_server.go (hunk 3 above)
Comment (alexgao001) [reply base/gfspapp/authenticator_server.go]: refined.

File: base/gfspapp/authenticator_server.go (hunk 2 above)
Comment (alexgao001) [reply base/gfspapp/authenticator_server.go]: refined.

File: base/gfspapp/authenticator_server.go (hunk 1 above)
Comment (alexgao001) [reply base/gfspapp/authenticator_server.go]: refined.

**Review by sysvm — APPROVED** [review]
LGTM

**Pull request merged**